script.js

```diff
@@ -0,0 +1,121 @@
+        return { area: area, figure: figure, input: { sideA: a, sideB: b, sideC: c }};
+        case 'square':
+        area = a ** 2;
+        return { area: area, figure: figure, input: {sideA: a}};
+        case 'rectangle':
+        area = a * b;
+        return { area: area, figure: figure, input: {sideA: a, sideB: b}};
+        default:
+        throw new Error("Wrong figure");
+    }
+};
+class Human {
+    constructor(name, age, dateOfBirth) {
+        this.name = name;
+        this.age = age;
+        this.dateOfBirth = dateOfBirth;
+    }
+    displayInfo() {
+        return `Name: ${this.name}, Age: ${this.age}, Date of birth: ${this.dateOfBirth}`;
+    }
+}
+class Employee extends Human {
+    constructor(name, age, dateOfBirth, salary, department) {
+        super(name, age, dateOfBirth);
+        this.salary = salary;
+        this.department = department;
+    }
+    displayInfo() {
+        return `${super.displayInfo()}, Salary: ${this.salary}, Department: ${this.department}`;
+    }
+}
+class Manager extends Employee {
+    constructor(name, age, dateOfBirth, salary, department, allTheDevs) {
+        super(name, age, dateOfBirth, salary, department);
+        if (!Array.isArray(allTheDevs) || allTheDevs == undefined) throw new Error("allTheDevs argument must present and must be an array");
+        this.allTheDevs = allTheDevs;
+    }
+    getDevs() {
+        let string = "";
+        this.allTheDevs.forEach((arg, i) => {
+            string = string + `Employee ${i+1} is ${arg}. \n`;
+        });
+        return string;
+    }
+    addDev(dev) {
+        this.allTheDevs.push(dev);
+    }
+    removeDev(index) {
+        this.allTheDevs.splice(index, 1);
+    }
+}
+class Developer extends Employee {
+    constructor(name, age, dateOfBirth, salary, department, manager) {
+        super(name, age, dateOfBirth, salary, department);
+        this.isManagedBy = manager;
+    }
+    changeManager(manager) {
+        this.isManagedBy = manager;
+    }
+    isManagedBy() {
+        return this.isManagedBy;
+    }
+}
+function *generator() {
+    yield {name: prompt("Ваше имя?")};
+    yield {lastname: prompt("Ваша фамилия?")};
+    yield {age: prompt("Ваш возраст?")};
+    yield {profession: prompt("Ваша профессия?")};
+}
+let profile = function() {
+    let obj = {};
+    let a = generator();
+    for (v of a) {
+        let key = Object.keys(v);
+        obj[key] = v[key];
+    }
+    console.log(obj);
+}
+profile();
+let createQueries = function() {
+    let promises = [];
+    rawUrl = "https://jsonplaceholder.typicode.com/users/";
+    for (let i = 1; i < 11; i++) {
+        let xhr = new XMLHttpRequest();
+        let promise = new Promise((res, rej) => {
+            let timeIsLeft = setTimeout(()=> {
+                rej("Rejecting...")
+            }, 20000);
+            xhr.open('GET', rawUrl+i);
+            xhr.send();
+            xhr.onreadystatechange = function() {
+                    if (xhr.status === 200) {
```

Comment:
```suggestion
                    if (xhr.readyState === xhr.DONE && xhr.status === 200) {
```
Надо проверять состояние объекта xhr, чтобы проводить ниже указанные операции над полностью загруженными данными